cloudstack/loadbalancer.go

```diff
@@ -1002,9 +1028,106 @@ func (lb *loadBalancer) createLoadBalancerRule(lbRuleName string, service *v1.Se
 		},
 	}
 
+	err = lb.updateLoadBalancerPool(lbRule, service)
+	if err != nil {
+		return nil, fmt.Errorf("updateLoadBalacerPool error: %v", err)
+	}
+
 	return lbRule, nil
 }
 
+func (lb *loadBalancer) updateLoadBalancerPool(lbRule *loadBalancerRule, service *v1.Service) error {
+
+	client, err := lb.getClient()
+	if err != nil {
+		return err
+	}
+
+	listGloboNetworkPoolsParams := &cloudstack.CustomServiceParams{}
+	listGloboNetworkPoolsResponse := globoNetworkPools{}
+
+	_, lbCustomHealthCheckVal := getLabelOrAnnotation(service.ObjectMeta, lbCustomHealthCheck)
+	if !lbCustomHealthCheckVal {
+		return nil
+	}
+
+	listGloboNetworkPoolsParams.SetParam("lbruleid", lbRule.Id)
+	err = client.Custom.CustomRequest("listGloboNetworkPools", listGloboNetworkPoolsParams, &listGloboNetworkPoolsResponse)
+
+	if err != nil {
+		return fmt.Errorf("error list load balancer pools for %v: %v", lbRule.Name, err)
+	}
+
+	if listGloboNetworkPoolsResponse.Count < 1 {
+		return nil
+	}
+
+	updateGloboNetworkPoolsParams := &cloudstack.CustomServiceParams{}
+
+	r := UpdateGloboNetworkPoolResponse{}
+	for idx := range service.Spec.Ports {
+		pool, err := generateGloboNetworkPool(idx, service, listGloboNetworkPoolsResponse.GloboNetworkPools)
+		if pool == (globoNetworkPool{}) {
+			continue
+		}
+		if err != nil {
+			return fmt.Errorf("error waiting for load balancer rule job %v: %v", lbRule.Name, err)
+		}
+		updateGloboNetworkPoolsParams.SetParam("poolids", pool.Id)
+		updateGloboNetworkPoolsParams.SetParam("lbruleid", lbRule.Id)
+		updateGloboNetworkPoolsParams.SetParam("healthchecktype", strings.ToUpper(pool.HealthCheckType))
+		updateGloboNetworkPoolsParams.SetParam("healthcheck", pool.HealthCheck)
+		updateGloboNetworkPoolsParams.SetParam("expectedhealthcheck", pool.HealthCheckExpected)
+		err = client.Custom.CustomRequest("updateGloboNetworkPool", updateGloboNetworkPoolsParams, &r)
+		if err != nil {
+			return fmt.Errorf("error updating globo network pool for %v: %v", lbRule.Name, err)
+		}
+		if r.JobID != "" {
+			err = waitJob(client, r.JobID, &r)
+			if err != nil {
+				return fmt.Errorf("error waiting for globo network pool for rule %v: %v", lbRule.Name, err)
+			}
+		}
+	}
+	return nil
+}
+
+func generateGloboNetworkPool(portsIdx int, service *v1.Service, globoPools []*globoNetworkPool) (globoNetworkPool, error) {
+	_, useTargetPort := getLabelOrAnnotation(service.ObjectMeta, lbUseTargetPort)
+	ports := service.Spec.Ports
+	dstPort := int(ports[portsIdx].NodePort)
+	vipPort := int(ports[portsIdx].Port)
+
+	if useTargetPort {
+		dstPort = ports[portsIdx].TargetPort.IntValue()
+	}
+
+	if ports[portsIdx].Name == "" || !strings.Contains(ports[portsIdx].Name, "-") {
+		return globoNetworkPool{}, nil
+	}
+
+	namedService := ports[portsIdx].Name
```

Comment:
```golang
namedPort := ports[portsIdx].Name
if namedPort == "" {
	return globoNetworkPool{}, nil
}

protocolParts := strings.Split(namedPort, "-")
if len(protocolParts) == 0 {
	return globoNetworkPool{}, nil
}

protocol := protocolParts[0]
```

Reply:
About `strings.Split`, actually according to documentation: `it returns If s does not contain sep and sep is not empty, Split returns a slice of length 1 whose only element is s.`,

Reply:
Sorry, I missed that. So, instead of zero we can use a less than 2 in the conditional block below the Split func.